jerry-core/ecma/base/ecma-errors.h

```diff
@@ -13,82 +13,27 @@
  * limitations under the License.
  */
 
-#include "config.h"
-
 #ifndef ECMA_ERRORS_H
 #define ECMA_ERRORS_H
 
-#if JERRY_ERROR_MESSAGES
-
-extern const char* const ecma_error_value_msg_p;
-extern const char* const ecma_error_wrong_args_msg_p;
-
-#if !JERRY_PARSER
-extern const char* const ecma_error_parser_not_supported_p;
-#endif /* !JERRY_PARSER */
-
-#if !JERRY_BUILTIN_JSON
-extern const char* const ecma_error_json_not_supported_p;
-#endif /* !JERRY_BUILTIN_JSON */
-
-#if !JERRY_ESNEXT
-extern const char* const ecma_error_symbol_not_supported_p;
-
-extern const char* const ecma_error_promise_not_supported_p;
-#endif /* !JERRY_ESNEXT */
-
-#if !JERRY_BUILTIN_TYPEDARRAY
-extern const char* const ecma_error_typed_array_not_supported_p;
-#endif /* !JERRY_BUILTIN_TYPEDARRAY */
-
-#if !JERRY_BUILTIN_SHAREDARRAYBUFFER
-extern const char* const ecma_error_shared_arraybuffer_not_supported_p;
-#endif /* !JERRY_BUILTIN_SHAREDARRAYBUFFER */
-
-#if !JERRY_BUILTIN_DATAVIEW
-extern const char* const ecma_error_data_view_not_supported_p;
-#endif /* !JERRY_BUILTIN_DATAVIEW */
-
-#if !JERRY_BUILTIN_BIGINT
-extern const char* const ecma_error_bigint_not_supported_p;
-#endif /* !JERRY_BUILTIN_BIGINT */
-
-#if !JERRY_BUILTIN_CONTAINER
-extern const char* const ecma_error_container_not_supported_p;
-#endif /* !JERRY_BUILTIN_CONTAINER */
-
-#if JERRY_MODULE_SYSTEM
-extern const char* const ecma_error_not_module_p;
-extern const char* const ecma_error_unknown_export_p;
-#else /* !JERRY_MODULE_SYSTEM */
-extern const char* const ecma_error_module_not_supported_p;
-#endif /* JERRY_MODULE_SYSTEM */
-
-extern const char* const ecma_error_callback_is_not_callable;
-extern const char* const ecma_error_arraybuffer_is_detached;
-extern const char* const ecma_error_cannot_convert_to_object;
-extern const char* const ecma_error_class_is_non_configurable;
-extern const char* const ecma_error_argument_is_not_an_object;
-extern const int ecma_error_argument_is_not_an_object_length;
-extern const char* const ecma_error_argument_is_not_a_proxy;
-extern const char* const ecma_error_target_is_not_a_constructor;
-extern const char* const ecma_error_argument_is_not_an_regexp;
-extern const char* const ecma_error_invalid_array_length;
-extern const char* const ecma_error_local_variable_is_redeclared;
-extern const char* const ecma_error_expected_a_function;
-
-#if JERRY_ESNEXT
-extern const char* const ecma_error_class_constructor_new;
-extern const char* const ecma_error_let_const_not_initialized;
-#endif /* JERRY_ESNEXT */
+#include "lit-globals.h"
 
+typedef enum
+{
+  ECMA_ERR_EMPTY,
+/** @cond doxygen_suppress */
+#if JERRY_ERROR_MESSAGES
+#define ECMA_ERROR_DEF(id, ascii_zt_string) id,
+#else /* !JERRY_ERROR_MESSAGES */
+#define ECMA_ERROR_DEF(id, ascii_zt_string) id = ECMA_ERR_EMPTY,
 #endif /* JERRY_ERROR_MESSAGES */
+#include "ecma-error-messages.inc.h"
+#undef ECMA_ERROR_DEF
+  /** @endcond */
+  ECMA_IS_VALID_CONSTRUCTOR
+} ecma_error_id_t;
```

Comment:
I don't really like the name for this, it does not identify the error, only the message. How about calling this `ecma_error_msg_t`